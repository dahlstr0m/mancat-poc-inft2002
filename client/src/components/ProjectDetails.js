// @flow

import * as React from 'react';
import { Component } from 'react-simplified';
import { ProjectCard, PosterCard } from './Card';
import { Alert } from './Widgets';
import {
  projectService,
  posterService,
  categoryService,
  employerService,
} from '../portfolio-service';

/**
 * Renders project details.
 */
export default class ProjectDetails extends Component<{ pathId: number }> {
  project: Project = {
    projectId: 0,
    title: '',
    projectDescription: '',
    projectDate: '',
    categoryId: 0,
    employerId: 0,
    active: false,
    ranking: 0,
  };
  category: Category = { categoryId: 0, categoryName: '' };
  employer: Employer = { employerId: 0, employerName: '' };
  posters: Poster[] = [];

  render() {
    return (
      <>
        <ProjectCard
          projectId={this.project.projectId}
          title={this.project.title}
          description={this.project.projectDescription}
          date={this.project.projectDate}
          category={this.category.categoryName}
          employer={this.employer.employerName}
        >
          {this.posters.map((poster) => (
            <PosterCard
              key={poster.posterId}
              posterId={poster.posterId}
              description={poster.posterDescription}
              url={poster.posterUrl}
              thumbnailUrl={poster.thumbnailUrl}
            >
              {'ingen data'}
            </PosterCard>
          ))}
        </ProjectCard>
      </>
    );
  }
  mounted() {
    projectService
      .getProject(this.props.pathId)
      .then((project) => (this.project = project))
      .catch((error: Error) => Alert.danger('Error getting project: ' + error.message));

    categoryService
      .getProjectCategory(this.props.pathId)
      .then((category) => (this.category = category))
      .catch((error: Error) => Alert.danger('Error getting category: ' + error.message));

    employerService
      .getProjectEmployer(this.props.pathId)
      .then((employer) => (this.employer = employer))
      .catch((error: Error) => Alert.danger('Error getting employer: ' + error.message));

    posterService
      .getProjectPosters(this.props.pathId)
      .then((posters) => (this.posters = posters))
      .catch((error: Error) => Alert.danger('Error getting posters: ' + error.message));
  }
}
