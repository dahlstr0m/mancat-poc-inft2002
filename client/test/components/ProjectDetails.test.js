// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

// Components
import ProjectDetails from '../../src/components/ProjectDetails';

// Mocks
jest.mock('../../src/services/project-service');
jest.mock('../../src/services/poster-service');
jest.mock('../../src/services/category-service');
jest.mock('../../src/services/employer-service');

// Tests
describe('ProjectDetails component tests', () => {
  test('ProjectDetails draws correctly', (done) => {
    const wrapper = shallow(<ProjectDetails pathId={1} />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();

      done();
    });
  });
});
