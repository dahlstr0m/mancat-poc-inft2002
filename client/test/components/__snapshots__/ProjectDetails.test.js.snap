// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails component tests ProjectDetails draws correctly 1`] = `
<Fragment>
  <Card
    title="Project 1"
  >
    <hr />
    <p>
      Project 1 description
    </p>
    <hr />
    <p>
      Kategori: 
      Kategori 1
    </p>
    <p>
      Oppdragsgiver: 
      Employer 1
    </p>
    <p>
      Prosjektdato: 
      10/10/2020
    </p>
  </Card>
  <CardGrid
    columns={1}
    columnsSm={2}
  />
</Fragment>
`;
